File: 2022/day_4/1.py
f = open("in.txt")
lines = f.read().split("\n")

total = 0
for line in lines:
    [a, b] = line.split(",")
    [a1, a2] = a.split("-")
    [b1, b2] = b.split("-")
    if (int(a1) >= int(b1) and int(a2) <= int(b2)) or (int(a1) <= int(b1) and int(a2) >= int(b2)):
        total += 1

print(total)

File: 2023/day_1/1.swift
import Foundation

func readFile(filename:String) -> String {
    var result = ""
    do {
        result = try String(contentsOfFile: "./"+filename, encoding: .utf8)
    }
    catch {
            print("ERROR: Cannot read file")
    }
    return result
}

let content = readFile(filename: "in.txt")
let lines = content.split(separator: "\n")

var total = 0
for line in lines {
    var first = -1
    var last = -1
    for c in line {
        let value = Int("\(c)") ?? -1
        if value > -1 {
            if first == -1 {
                first = value
            }
            last = value
        }
    }
    total = total + 10 * first + last
}
print(total)
